packages/core/src/storage.ts

```diff
@@ -208,6 +208,42 @@ export type FindResult<T extends Doc> = WithLookup<T>[] & {
 // eslint-disable-next-line @typescript-eslint/no-empty-interface
 export interface TxResult {}
 
+/**
+ * @public
+ */
+export interface IndexedDoc {
+  id: Ref<Doc>
+  _class: Ref<Class<Doc>>
+  space: Ref<Space>
+  modifiedOn: Timestamp
+  modifiedBy: Ref<Account>
+  attachedTo?: Ref<Doc>
+  attachedToClass?: Ref<Class<Doc>>
+  [key: string]: any
+}
+
+/**
+ * @public
+ */
+export type FulltextQuery = any
+
+/**
+ * @public
+ */
+export type FulltextQueryOptions = any
+
+/**
+ * @public
+ */
+export interface FulltextSearchResult {
+  suggest?: any
```

Comment:
May we elaborate on this a bit more?

Reply:
Do you mean naming or the structure of results?